projects/components/src/filtering/filter/parser/types/abstract-filter-parser.ts

```diff
@@ -0,0 +1,66 @@
+import { FilterAttribute } from '../../filter-attribute';
+import { FilterAttributeType } from '../../filter-attribute-type';
+import { FilterOperator } from '../../filter-operators';
+import { ParsedFilter } from '../parsed-filter';
+
+export abstract class AbstractFilterParser<TValue> {
+  public static supportedAttributeTypes: FilterAttributeType[];
+  public static supportedOperators: FilterOperator[];
+
+  protected abstract supportedAttributeTypes(): FilterAttributeType[];
+  protected abstract supportedOperators(): FilterOperator[];
+
+  protected abstract parseValueString(attribute: FilterAttribute, valueString: string): TValue | undefined;
+
+  public parseFilterString(attributes: FilterAttribute[], filterString: string): ParsedFilter<TValue> | undefined {
+    const splitFilter = this.splitFilterString(filterString);
+
+    if (splitFilter === undefined) {
+      return undefined;
+    }
+
+    const attribute = attributes.find(a => a.name.toLowerCase() === splitFilter.key.toLowerCase());
+
+    if (attribute === undefined) {
+      return undefined;
+    }
+
+    const parsedValue: TValue | undefined = this.parseValueString(attribute, splitFilter.value);
+
+    if (parsedValue === undefined) {
+      return undefined;
+    }
+
+    return {
+      key: attribute.name,
+      operator: splitFilter.operator,
+      value: parsedValue
+    };
+  }
+
+  protected splitFilterString(filterString: string): ParsedFilter<string> | undefined {
+    const matchingOperator = this.supportedOperators()
+      .sort((o1: string, o2: string) => o2.length - o1.length) // Sort on length so we check multichar first
+      .map(op => ` ${op} `)
+      .find(op => filterString.includes(op));
+
+    if (matchingOperator === undefined) {
+      return undefined;
+    }
+
+    const parts = filterString
+      .split(matchingOperator)
+      .map(str => str.trim())
+      .filter((part: string | undefined) => (part ?? '').length > 0);
```

Comment:
how could this be undefined? we invoke trim the line above, so we'd have already errored out

Reply:
Ah, true. I'll fix locally for now unless you want this addressed before merge.

Reply:
No need